File: src/test/java/uk/gov/justice/hmpps/prison/repository/v1/NomisV1SQLErrorCodeTranslatorTest.java
package uk.gov.justice.hmpps.prison.repository.v1;

import org.junit.jupiter.api.Test;

import java.sql.SQLException;

import static org.assertj.core.api.Assertions.assertThat;

public class NomisV1SQLErrorCodeTranslatorTest {
    private final NomisV1SQLErrorCodeTranslator translator = new NomisV1SQLErrorCodeTranslator();

    @Test
    public void customTranslate_errorCode() {
        final var accessException = translator.customTranslate("hello", "sql", new SQLException("reason", "state", 20040));
        assertThat(accessException).hasMessage("Sum of sub account balances not equal to current balance");

    }

    @Test
    public void customTranslate_notMapped() {
        final var accessException = translator.customTranslate("hello", "sql", new SQLException("reason"));
        assertThat(accessException).isNull();
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/repository/v1/NomisV1SQLErrorCodeTranslatorTest.kt
package uk.gov.justice.hmpps.prison.repository.v1

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.sql.SQLException

class NomisV1SQLErrorCodeTranslatorTest {
  private val translator = NomisV1SQLErrorCodeTranslator()

  @Test
  fun customTranslate_errorCode() {
    val accessException = translator.customTranslate("hello", "sql", SQLException("reason", "state", 20040))
    assertThat(accessException)
      .hasMessage("Sum of sub account balances not equal to current balance")
  }

  @Test
  fun customTranslate_notMapped() {
    val accessException = translator.customTranslate("hello", "sql", SQLException("reason"))
    assertThat(accessException).isNull()
  }
}
